libc/src/stdio/scanf_core/CMakeLists.txt

```
@@ -62,6 +62,7 @@ add_object_library(
     reader.h
   DEPENDS
     libc.src.__support.macros.attributes
+    libc.hdr.types.FILE
```

This is missing the dependency on `libc.src.__support.File.file`, as well as the flag controlling if it's using the system file. You'll also need to set up proper handling for the GPU like what `vfscanf_internal` has.
```suggestion
    libc.src.__support.File.file
    libc.hdr.types.FILE
  ${use_system_file}
```

---

I seem to be getting an error like (after adding the above change):
```
  The dependency target "libc.src.__support.File.file" of target
  "libc.src.stdio.scanf_core.reader" does not exist.
```

is that expected on a M1 mac?

using the build commands from:https://github.com/llvm/llvm-project/issues/115394#issuecomment-2574165429

---

ah, that's because we only support overlay mode on mac, and overlay mode doesn't have `FILE`. You'll need to add logic to only add `libc.src.__support.File.file` to the list of `DEPENDS` if `LLVM_LIBC_FULL_BUILD` is true.

---

>  like what vfscanf_internal has.

So, I think there's a bug / typo in the CMake call there.

Instead of:

```
  add_header_library(
    vfscanf_internal
    HDRS
      vfscanf_internal.h
    DEPENDS
      .reader
      .scanf_main
      libc.include.stdio
      libc.src.__support.File.file
      libc.src.__support.arg_list
    ${use_system_file}
  )
```

it should be:

```
  add_header_library(
    vfscanf_internal
    HDRS
      vfscanf_internal.h
    DEPENDS
      .reader
      .scanf_main
      libc.include.stdio
      libc.src.__support.File.file
      libc.src.__support.arg_list
    COMPILE_OPTIONS
      ${use_system_file}
  )
```

I think is what caused my build to have the bugs with macOS' system stdio.h macros

---

The `COMPILE_OPTIONS` shouldn't be necessary, because `use_system_file` already contains it, see: https://github.com/llvm/llvm-project/blob/main/libc/src/stdio/CMakeLists.txt#L30